src/v/cluster/rm_stm.cc

```diff
@@ -1809,8 +1812,14 @@ ss::future<raft::stm_snapshot> rm_stm::do_take_local_snapshot(uint8_t version) {
             stm_snapshot.producers.push_back(std::move(snapshot));
         }
     }
```

Comment:
I can see this is the only STM we hold the units across scheduling points, potentially holding back applies. Is it for a reason, i.e. the operations we do above conflict with the ones we do in apply? Or is it only because of the code structure?

Reply:
thats a good question, I was pondering too and I think its mostly for historical reasons and no one really bothered to touch this sensitive piece of code, it can probably rewritten to avoid that.

Reply:
are there any plans to benchmark how this change affects performance?

Reply:
hopefully once it lands, we can do the omb transactions workload that Michal added recently.